go/controller/internal/httpserver/handlers/modelconfig.go

```diff
@@ -60,10 +137,499 @@ func (h *ModelConfigHandler) HandleGetModelConfig(w ErrorResponseWriter, r *http
+		ModelParams:      modelParams,
+	}
+
+	log.Info("Successfully retrieved and formatted model config")
+	RespondWithJSON(w, http.StatusOK, responseItem)
+}
+
+// Helper function to get all JSON keys from a struct type
+func getStructJSONKeys(structType reflect.Type) []string {
+	keys := []string{}
+	if structType.Kind() != reflect.Struct {
+		return keys
+	}
+	for i := 0; i < structType.NumField(); i++ {
+		field := structType.Field(i)
+		jsonTag := field.Tag.Get("json")
+		if jsonTag != "" && jsonTag != "-" {
+			tagParts := strings.Split(jsonTag, ",")
+			keys = append(keys, tagParts[0])
+		}
+	}
+	return keys
+}
+
+// Helper function to get JSON keys specifically marked as required
+func getRequiredKeys(providerType v1alpha1.ModelProvider) []string {
+	switch providerType {
+	case v1alpha1.AzureOpenAI:
+		// Based on the +required comments in the AzureOpenAIConfig struct definition
+		return []string{"azureEndpoint", "apiVersion"}
+	case v1alpha1.OpenAI, v1alpha1.Anthropic, v1alpha1.Ollama:
+		// These providers currently have no fields marked as strictly required in the API definition
+		return []string{}
+	default:
+		// Unknown provider, return empty
+		return []string{}
+	}
+}
+
+func (h *ModelConfigHandler) HandleListSupportedProviders(w ErrorResponseWriter, r *http.Request) {
+	log := ctrllog.FromContext(r.Context()).WithName("modelconfig-handler").WithValues("operation", "list-supported-providers")
+
+	log.Info("Listing supported providers with parameters")
+
+	providersData := []struct {
+		providerEnum v1alpha1.ModelProvider
+		configType   reflect.Type
+	}{
+		{v1alpha1.OpenAI, reflect.TypeOf(v1alpha1.OpenAIConfig{})},
+		{v1alpha1.Anthropic, reflect.TypeOf(v1alpha1.AnthropicConfig{})},
+		{v1alpha1.AzureOpenAI, reflect.TypeOf(v1alpha1.AzureOpenAIConfig{})},
+		{v1alpha1.Ollama, reflect.TypeOf(v1alpha1.OllamaConfig{})},
+	}
+
+	providersResponse := []map[string]interface{}{}
+
+	for _, pData := range providersData {
+		allKeys := getStructJSONKeys(pData.configType)
+		requiredKeys := getRequiredKeys(pData.providerEnum)
+		requiredSet := make(map[string]struct{})
+		for _, k := range requiredKeys {
+			requiredSet[k] = struct{}{}
+		}
+
+		optionalKeys := []string{}
+		for _, k := range allKeys {
+			if _, isRequired := requiredSet[k]; !isRequired {
+				optionalKeys = append(optionalKeys, k)
+			}
+		}
+
+		providersResponse = append(providersResponse, map[string]interface{}{
+			"name":           string(pData.providerEnum),
+			"type":           string(pData.providerEnum),
+			"requiredParams": requiredKeys,
+			"optionalParams": optionalKeys,
+		})
+	}
+
+	RespondWithJSON(w, http.StatusOK, providersResponse)
+}
+
+type CreateModelConfigRequest struct {
+	Name            string                      `json:"name"`
+	Provider        Provider                    `json:"provider"`
+	Model           string                      `json:"model"`
+	APIKey          string                      `json:"apiKey"`
```

Comment:
Should this be optional here as well. Allow the user to point to an existing secret?

Reply:
will do in a separate PR -- allow the users to pick an existing secret vs. create a new one